packages/typespec-python/src/emitter.ts

```diff
@@ -299,7 +299,12 @@ function emitParameter(
         type = type["valueType"];
     }
     const paramMap: Record<string, any> = {
-        wireName: parameter.type === "path" ? parameter.param.name : parameter.name,
+        wireName:
+            parameter.type === "path"
+                ? parameter.param.name
+                : parameter.type === "query" && parameter.name.toLocaleLowerCase() === "apiversion"
+                ? "api-version"
+                : parameter.name,
```

Comment:
According to [Azure api guideline](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#versioning-api-version-query-param), its wire name must be "api-version". Since [cadl-ranch case](https://github.com/Azure/cadl-ranch/blob/ab1a6073cfc661811d5dcc8de787007a9406021e/packages/cadl-ranch-specs/http/server/versions/versioned/main.tsp#L51) doesn't define it with real wire name like [this case](https://github.com/Azure/cadl-ranch/blob/ab1a6073cfc661811d5dcc8de787007a9406021e/packages/cadl-ranch-specs/http/server/versions/not-versioned/main.tsp#L32), emitter has to do the additional logic.

Reply:
After off-line discussion, we think it is better to update cadl-ranch test case.